Sources/Core/Sources/ECS/Singles/InputState.swift

```diff
@@ -129,9 +121,11 @@ public final class InputState: SingleComponent {
     }
 
     for event in newlyReleased {
-      ///test for forwards key being released
+      // test for forwards key being released
       if event.input == .moveForward {
-        /// if the forwards key has been released, set the time since it was released
+        // if the forwards key has been released, stop sprinting
+        inputs.remove(.sprint)
+        // if the forwards key has been released, set the time since it was released
         forwardsUpTime = tickCount
```

Comment:
I just checked the Minecraft source code (decompiled), and it looks like we actually need to track the time between pressing the key down and pressing it down again, not the time between releasing the key and pressing it the next time.

This makes sense because otherwise if you're walking for however long and then quickly release and repress the forwards key, then you start sprinting.